Sources/Automerge/patch.swift

```diff
@@ -4,10 +4,29 @@ import enum AutomergeUniffi.PatchAction
 typealias FfiPatchAction = AutomergeUniffi.PatchAction
 typealias FfiPatch = AutomergeUniffi.Patch
 
+/// A collection of updates applied to an Automerge document.
+///
+/// A patch can be received from applying updates to an Automerge document with one of the following methods:
+/// -  ``Document/applyEncodedChangesWithPatches(encoded:)``
+/// - ``Document/receiveSyncMessageWithPatches(state:message:)``
+/// - ``Document/mergeWithPatches(other:)``.
+///
+/// You can inspect these patches to identify the objects updated within the Automerge document, in order to react accordingly within your code.
+/// A common use case for inspecting patches is to recalculate derived data that is using Automerge as an authoritative source.
 public struct Patch: Equatable {
+    /// The kind of update to apply.
     public let action: PatchAction
+
+    /// The path to the object identifier the update effects.
     public let path: [PathElement]
 
+    /// Creates a new patch
+    /// - Parameters:
+    ///   - action: The kind of update to apply.
+    ///   - path: The path to the object identifier that the action effects.
+    ///
+    ///   The `path` does not identify the property on an object, or index in a sequence, that is updated, only the object that is effected.
+    ///   The `action` includes the type of change, and the value being updated, if relevant to the change.
     public init(action: PatchAction, path: [PathElement]) {
```

Comment:
I recall we discussed making this `internal` as this would never be constructed by the user. Can we do that?

Reply:
Done in the follow-up commit, left the comments on the initializer anyway - but made it 'protected' so it's not accessible external to the module (i.e. developers using the package).